### ISSUE TITLE
Support creating of Series with python Objects natively in rust.

### ISSUE THREAD

**Issue by ghuls** [pola-rs/polars#1897]
#### Describe your feature request

While debugging some PyPy test failures, I discovered that this test fails:

```python
def test_object() -> None:
    vals = [[12], "foo", 9]
    a = pl.Series("a", vals)
    assert a.dtype == pl.Object
    assert a.to_list() == vals
    assert a[1] == "foo"
```

As it relies on pyarrow.

As now storing Python objects inside Arrow Series is supported, it might be good to support creation of Object series natively in rust too.

```python
In [39]: import polars as pl

In [40]: pl.internals.construction._PYARROW_AVAILABLE = False

In [41]: vals = [[12], "foo", 9]

In [42]: a = pl.Series("a", vals)
thread '<unnamed>' panicked at 'called `Result::unwrap()` on an `Err` value: PyErr { type: <class 'TypeError'>, value: TypeError("'str' object cannot be interpreted as an integer"), traceback: None }', src/list_construction.rs:26:53
---------------------------------------------------------------------------
PanicException                            Traceback (most recent call last)
<ipython-input-42-1d5d56e454ea> in <module>
----> 1 a = pl.Series("a", vals)

~/software/anaconda3/envs/polars/lib/python3.8/site-packages/polars/internals/series.py in __init__(self, name, values, dtype, strict, nan_to_null)
    207             self._s = numpy_to_pyseries(name, values, strict, nan_to_null)
    208         elif isinstance(values, Sequence):
--> 209             self._s = sequence_to_pyseries(name, values, dtype=dtype, strict=strict)
    210         elif _PANDAS_AVAILABLE and isinstance(values, (pd.Series, pd.DatetimeIndex)):
    211             self._s = pandas_to_pyseries(name, values)

~/software/anaconda3/envs/polars/lib/python3.8/site-packages/polars/internals/construction.py in sequence_to_pyseries(name, values, dtype, strict)
    132             if not _PYARROW_AVAILABLE:  # pragma: no cover
    133                 dtype = py_type_to_dtype(nested_dtype)
--> 134                 return PySeries.new_list(name, values, dtype)
    135 
    136             try:

PanicException: called `Result::unwrap()` on an `Err` value: PyErr { type: <class 'TypeError'>, value: TypeError("'str' object cannot be interpreted as an integer"), traceback: None }

In [43]: pl.internals.construction._PYARROW_AVAILABLE = True

In [44]: a = pl.Series("a", vals)

In [45]: a
Out[45]: 
shape: (3,)
Series: 'a' [o][object]
[
	[12]
	foo
	9
]

In [46]: type(a[0])
Out[46]: list

In [47]: type(a[1])
Out[47]: str

In [48]: type(a[2])
Out[48]: int
```

**Comment by ritchie46** [comment]
I don't think it relies on `pyarrow`. Pyarrow has no concept of the polars `Object` datatype. The issue is that we have several `try except` branches and if all fails, we create a `PySeries.from_object`.

It means that we need to wrap the `PySeries.from_list` into a `try except` branch as well. 

Could you make a PR?

**Comment by ritchie46** [comment]
closed #1906
